### PR TITLE
LINDA

### PR DESCRIPTION
**Pull request by tigercat2000**
This ports LINDA from TG, as well as their firedoor system, and a slightly new on_fire system.

# Very-very-very much DO NOT MERGE.

There are quite a few bugs still to track down and work out-
 - [X] Make atmos stop phasing through closed doors EDIT: ofc airlocks don't call parent on close(), and the first version of LINDA didn't put the right proc in it. ;-;
 - [X] Delete FEA
 - [x] This PR needs more awesome screenshots EDIT: oh yay found some
 - [X] Breathe, I haven't done it in like a few days Edit: okay I slept and I think I breathed
 - [X] I think away missions need their firelocks replaced. Edit: Nope, they don't need them replaced.

Awesome screenshot
More awesome
ded

### CODE REVIEW

**Comment by Spacemanspark** [comment]
I really still wish we kept ZAS.
However, this will make plasma grief slightly less devastating.
Soooooo...
:+1: :-1:

**Comment by ZomgPonies** [comment]
:+1:  :100: :+1:

**Comment by Fox-McCloud** [comment]
:+1: x :100: x :100:

**Comment by TheDZD** [comment]
Thumbs Up :heavy_multiplication_x:

**Comment by fj45** [comment]
*scream

*shout

*flip

*scream

**Comment by Keilden** [comment]
So with this system gas like plasma won't fill an entier room at once? and there must be enough of it to fill the room in first place.

**Comment by Regen1** [comment]
Is there a thread about this on the forums? If no then there should be, this is a HUGE change and the community should have a say inn it

**Comment by ZomgPonies** [comment]
This is a change that we've been discussing and confirming we've been wanting to do for over a year now. No there isn't any thread, feel free to start one but it isn't up to community vote, this is what we want from our atmos system.

**Comment by fj45** [comment]
Yeah, to put it simply, ZAS is completely broken, LINDA is not. LINDA also allows for better chemical reactions and far better fires.

File: code/modules/assembly/igniter.dm [hunk 1]
```diff
@@ -22,7 +22,7 @@
 			if (istype(src.loc.loc, /obj/item/weapon/reagent_containers/glass/beaker/))
 				var/obj/item/weapon/reagent_containers/glass/beaker/beakerbomb = src.loc.loc
 				if(beakerbomb)
-					beakerbomb.ignite()
+					beakerbomb.fire_act()
```
Comment (Fox-McCloud) [inline code/modules/assembly/igniter.dm]: this is supposed to be ignite--as it's a special proc only for beakers--if you need to rename it, that's fine, but it's definitely not supposed to be fire_act

File: code/modules/reagents/reagent_containers/glass_containers.dm [hunk 2]
```diff
@@ -246,7 +246,7 @@
 	else
 		usr << "<span class='notice'>There is no assembly to remove.</span>"
 
-/obj/item/weapon/reagent_containers/glass/beaker/ignite()
+/obj/item/weapon/reagent_containers/glass/beaker/fire_act()
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/glass_containers.dm]: this is supposed to be ignite--as it's a special proc only for beakers--if you need to rename it, that's fine, but it's definitely not supposed to be fire_act

Comment (tigercat2000) [inline code/modules/reagents/reagent_containers/glass_containers.dm]: ignite wasn't a special proc for beakers, though, it was a global /atom/ proc

Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/glass_containers.dm]: oops.

**Comment by tigercat2000** [comment]
Preemptively fixed merge conflicts with ParadiseSS13/Paradise#900, that will have to be merged before this is.

**Comment by ZomgPonies** [comment]
:fu: ZAS

File: code/modules/reagents/newchem/pyro.dm [hunk 3]
```diff
@@ -38,14 +38,14 @@
 /datum/reagent/clf3/on_mob_life(var/mob/living/M as mob)
 	if(!M) M = holder.my_atom
 	M.adjust_fire_stacks(4)
-	M.adjustFireLoss(0.35*M.fire_stacks)
+	M.adjustFireLoss(0.35*1)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/pyro.dm]: this should defnitely be M.fire_stacks; it does burn damage based on how many fire stacks the mob has.

Comment (tigercat2000) [inline code/modules/reagents/newchem/pyro.dm]: aahh I knew I forgot to reapply firestacks somewhere

**Comment by Fox-McCloud** [comment]
oddly enough spacewind is still throwing you 5-7 tiles, even after your update--it's strange.

I noticed atmos is processing super fast; in a particular enclosed space it took about 60 seconds to go from 101 to 22 kPa when that area was exposed to space (on TG)---on this implementation, that happened in about 10 seconds.

**Comment by tigercat2000** [comment]
@Fox-McCloud -tg-'s atmos runs every 2 seconds, give or take, ours runs about every second. I'll need details about the spacewind, I couldn't recreate the bug on a local server.

**Comment by Regen1** [comment]
Did you fix the bug where gasses are not actually being breathed by mobs?

**Comment by tigercat2000** [comment]
@Regen1 Yes, commit 173bdc4c2763140beda1644490860e2bfee96013 fixes mob breathing.

**Comment by fj45** [comment]
gif

**Comment by PJB3005** [comment]
And from that day on, paradise lagged even more than before.

**Pull request merged**